packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+class WindowsVersionValidator extends DoctorValidator {
+  /// Validator to be run with `flutter doctor` to check
+  /// Windows host machines if they are running supported versions,
+  /// current unsupported versions = 7, 8
+  const WindowsVersionValidator({required ProcessManager processManager})
+      : _processManager = processManager,
+        super('Windows Version');
+
+  final ProcessManager _processManager;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result;
+    try {
+      result = _processManager.runSync(<String>['systeminfo']);
+    } on ProcessException {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo:
+            'Unable to run Windows version check using built-in `systeminfo`',
+      );
+    }
+
+    if (result.exitCode != 0) {
```

Comment:
Same here

Reply:
Do you know in what cases `systeminfo` will return non-zero?

Reply:
I can't find any examples of why this would return non-zero but just wanted to cover all basis because the following code would rely on a successful exit code